src/thor/route_action.cc

```diff
@@ -356,7 +376,8 @@ void thor_worker_t::path_arrive_by(Api& api, const std::string& costing) {
   for (auto origin = ++correlated.rbegin(); origin != correlated.rend(); ++origin) {
     // Get the algorithm type for this location pair
     auto destination = std::prev(origin);
-    thor::PathAlgorithm* path_algorithm = get_path_algorithm(costing, *origin, *destination);
+    thor::PathAlgorithm* path_algorithm =
+        get_path_algorithm(costing, *origin, *destination, api.options());
```

Comment:
needed to pass the options so that the algorithm selection could check the date_time_type